### PR TITLE
refactor(platonic/regress): two-step reviewer-friendly pipeline

### PR DESCRIPTION
**Pull request by ksd3**
## Summary

Collapses the regression tier into one cooperative worker + one local aggregator, with the published preprocessing baked into the worker (no more "fix-up" re-probe step). A reviewer can now reproduce the regression numbers end to end with two scripts and four environment variables.

Reviewer reproduction:
\`\`\`bash
export HF_TOKEN=... PU_REGRESS_RESULTS_REPO=<owner>/results PU_EMB_REPO=<owner>/embeddings
python setup_hf.py "\$PU_REGRESS_RESULTS_REPO"
bash vast/vast_regress.sh                                        # or sbatch slurm/...
python 02_aggregate_pairs.py --pull-from "\$PU_REGRESS_RESULTS_REPO" --out-dir ./derived
\`\`\`

## Probe recipe (now in \`01_extract_and_probe.py\`)

- redshift only: filter \`z > 0\`, clip to \`[0, 4]\`
- all properties: 1–99% quantile clip on labels
- StandardScaler fit on the train fold, applied to test fold
- 10 random 80/20 splits, mean ± std R²
- GPU \`torch.linalg.lstsq\` (sklearn CPU fallback)

These are the values that match Mike's \`r2_vs_params_45000galaxies_upsampled.json\` reference (mean abs diff 0.006, max 0.029 across 150 cells).

## Changes

- Renamed \`pu_regress.py\` → \`01_extract_and_probe.py\`, \`pu_regress_pairs.py\` → \`02_aggregate_pairs.py\`
- Patched probe section: dropped CV / permutation / PCA, added \`_prep_property\` and \`_probe_published\`
- Added optional \`.npy\` upload to \`\$PU_EMB_REPO\` so downstream re-probing doesn't need a GPU
- Rewrote README around the explicit two-step flow + recipe + per-knob defaults
- Updated filename references in \`setup_hf.py\`, \`slurm/pu_regress.sh\`, \`vast/vast_regress.sh\`

## Test plan

- [x] Both scripts parse (\`python -c "import ast; ast.parse(open(p).read())"\`)
- [x] Re-ran the probe locally on cached embeddings via the new code path; values agree with Mike's reference within 0.006 mean / 0.029 max
- [ ] @shash12-ship — quick read-through of \`01_extract_and_probe.py\` probe block + README before merge

cc @shash12-ship

### CODE REVIEW

**Comment by ksd3** [comment]
@shash12-ship — pinging for a quick review when you have a moment. Main thing to scan is the new probe block in `01_extract_and_probe.py` (`_prep_property` + `_probe_published`) and the rewritten README. Output values verified against the published reference numbers.

**Comment by ksd3** [comment]
Added step 3: `03_plot_umap.py`. Generated locally on the existing `umap.parquet` files — produces `umap_hsc.pdf` and `umap_jwst.pdf`, each ~67 MB, one page per coloured-by physics property, grid = family × size with a shared random subsample for fair cross-panel comparison.

Pushed in 9cc2e21.

**Review by Shash12-ship — APPROVED** [review]
